### ISSUE TITLE
interspersed autofix of multiple expressions on same line

### ISSUE THREAD

**Issue by gordonwoodhull** [semgrep/semgrep#3577]
Similar to #3388, if fixing multiple expressions in the same line, the autofixes can get mixed up and duplicated, resulting in garbled code.

Here is the repro: https://semgrep.dev/s/gordonwoodhull:wrap-strings

On semgrep.dev it produces the correct individual suggestions. However, when running semgrep 0.59 on the command line with `--autofix`, it looks like it confuses the output positions between the two fixes.

The rule attempts to replace the expression pattern `(String $S)` with the fix `wrap($S)`.

On input

```java
        return "a" + "b";
```

it produces the output

```java
        return wrap("wrap("b") + "b";
```

**Comment by emjin** [comment]
I assume this is also P1 for you?

**Comment by gordonwoodhull** [comment]
Yes. I would say it is P1 because we can still use Semgrep for a lot of things but this makes a small percentage of our fixes impossible.

(We are trying to do some pretty ambitious things with autofix! Thank you for the amazing tool.)

**Comment by emjin** [comment]
Glad to hear that :D Autofix is still an experimental feature, but we're really excited about what people can do with it and hope we can put it on our roadmap soon.

**Comment by gordonwoodhull** [comment]
I am planning to work on this.

**Comment by gordonwoodhull** [comment]
Outline of solution in https://github.com/returntocorp/semgrep/issues/3388#issuecomment-967756894.

Hope to contribute soon.

**Comment by ievans** [comment]
Confirmed fixed in 0.77! Playground is confused now but semgrep CLI is solid.

**Comment by gordonwoodhull** [comment]
Thanks @chair6! Took a glance at your solution and it looks equivalent and maybe simpler than what I did.

Will check for edge cases soon.
